extension/indexer/tx_indexer.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/fees"
+)
+
+var (
+	_ AcceptedSubscriber = (*txDBIndexer)(nil)
+	_ AcceptedSubscriber = (*NoOpTxIndexer)(nil)
+
+	failureByte = byte(0x0)
+	successByte = byte(0x1)
+)
+
+type TxIndexer interface {
+	AcceptedSubscriber
+	GetTransaction(txID ids.ID) (bool, int64, bool, fees.Dimensions, uint64, error)
+}
+
+type txDBIndexer struct {
+	db database.Database
+}
+
+func NewTxDBIndexer(db database.Database) *txDBIndexer {
+	return &txDBIndexer{db: db}
+}
+
+func (i *txDBIndexer) Accepted(_ context.Context, blk *chain.StatelessBlock) error {
```

Comment:
nit: export the struct + rename the receiver

Reply:
Done